Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
sorry - let me phrase this differently, because i still don't understand. 

what is special about `decodableArrayForKey ` and `enumArrayForKey ` that the check for the `ArrayPartiallyDecodable` type whereas the other array functions do not?